src/Data/Food/Retail.elm

```diff
@@ -0,0 +1,193 @@
+        { energy = rate (kilowattHours 61.54) (cubicMeters 1)
+        , cooling = rate (kilowattHours 415.38) (cubicMeters 1)
+        , water = ratio (liters 280.8) (cubicMeters 1)
+        , transport = Length.kilometers 600
+        }
+
+
+displayNeeds : Conservation -> String
+displayNeeds (Conservation type_ _) =
+    case type_ of
+        Ambient ->
+            -- FIXME Would be better to display the rounded floats above
+            "Énergie: 123.08 kWh/m³, Réfrigération: 0 kWh/m³, Eau 561.5 L/m³, Transport: 600km"
+
+        Chilled ->
+            "Énergie: 46.15 kWh/m³, Réfrigération: 219.23 kWh/m³, Eau 210.6 L/m³, Transport: 600km"
+
+        Frozen ->
+            "Énergie: 61.54 kWh/m³, Réfrigération: 415.38 kWh/m³, Eau 280.8 L/m³, Transport: 600km"
+
+
+all : List Conservation
+all =
+    -- for selection list in the builder
+    [ ambient, chilled, frozen ]
+
+
+toString : Conservation -> String
+toString (Conservation type_ _) =
+    case type_ of
+        Ambient ->
+            "ambient"
+
+        Chilled ->
+            "chilled"
+
+        Frozen ->
+            "frozen"
+
+
+fromString : String -> Result String Conservation
+fromString str =
+    case str of
+        "ambient" ->
+            Ok ambient
+
+        "chilled" ->
+            Ok chilled
+
+        "frozen" ->
+            Ok frozen
+
+        _ ->
+            Err "Type de conservation incorrect"
+
+
+toDisplay : Conservation -> String
+toDisplay (Conservation t _) =
+    case t of
+        Ambient ->
+            "Sec"
+
+        Chilled ->
+            "Frais"
+
+        Frozen ->
+            "Surgelé"
+
+
+encode : Conservation -> Encode.Value
+encode =
+    Encode.string << toString
+
+
+decode : Decoder Conservation
+decode =
+    Decode.string
+        |> Decode.andThen (fromString >> RE.unpack Decode.fail Decode.succeed)
+
+
+waterImpact : Float -> Volume -> Process -> Impacts
+waterImpact waterNeeds volume =
+    .impacts
+        >> Impact.mapImpacts
+            (\_ impact ->
+                impact
+                    |> Unit.impactToFloat
+                    |> (*) (Quantity.multiplyBy waterNeeds volume |> Volume.inLiters)
+                    |> Unit.impact
+            )
+
+
+elecImpact : Quantity Float (Rate Joules CubicMeters) -> Volume -> Process -> Impacts
+elecImpact elecNeeds volume =
+    .impacts
+        >> Impact.mapImpacts
+            (\_ impact ->
+                impact
+                    |> Unit.impactToFloat
+                    |> (*) (Quantity.at elecNeeds volume |> Energy.inKilowattHours)
+                    |> Unit.impact
+            )
+
+
+transportImpact : Length -> Mass -> Process -> Impacts
+transportImpact distance mass =
+    .impacts
+        >> Impact.mapImpacts
+            (\_ impact ->
+                impact
+                    |> Unit.impactToFloat
+                    |> (*) (Length.inKilometers distance * Mass.inMetricTons mass)
+                    |> Unit.impact
+            )
```

Comment:
Have you considered extracting this to some dedicated generic helper?

Reply:
Actually the generic helper is already the Impacts.mapImpacts function
(which should even be shortened as Impacts.map, since it's just an AnyDict.map)